Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -224,7 +224,7 @@ TextureCacheBase::TCacheEntryBase* TextureCacheBase::DoPartialTextureUpdates(Tex
 
 	TexCache::iterator iter = textures_by_address.lower_bound(entry_to_update->addr);
 	TexCache::iterator iterend = textures_by_address.upper_bound(entry_to_update->addr + entry_to_update->size_in_bytes);
-	bool entry_need_scaling = true;
+	bool entry_need_scaling = g_ActiveConfig.bCopyEFBScaled;
```

Comment:
I think both ways are wrong a bit. In thery, there can be two entrys with different scaling factors. But I'm not sure if we want to support this case. But we should support to only upscale if the entry is upscaled at all.